src/express-lane-auction/ExpressLaneAuction.sol

```diff
@@ -0,0 +1,538 @@
+        (ELCRound storage lastRoundResolved, ) = latestResolvedRounds.latestELCRound();
+        if (roundTimingInfo.isReserveBlackout(lastRoundResolved.round)) {
+            revert ReserveBlackout();
+        }
+
+        _setReservePrice(newReservePrice);
+    }
+
+    /// @inheritdoc IExpressLaneAuction
+    function balanceOf(address account) external view returns (uint256) {
+        return _balanceOf[account].balanceAtRound(roundTimingInfo.currentRound());
+    }
+
+    /// @inheritdoc IExpressLaneAuction
+    function withdrawableBalance(address account) external view returns (uint256) {
+        return _balanceOf[account].withdrawableBalanceAtRound(roundTimingInfo.currentRound());
+    }
+
+    /// @inheritdoc IExpressLaneAuction
+    function deposit(uint256 amount) external {
+        _balanceOf[msg.sender].increase(amount);
+        biddingToken.safeTransferFrom(msg.sender, address(this), amount);
+        emit Deposit(msg.sender, amount);
+    }
+
+    /// @inheritdoc IExpressLaneAuction
+    function initiateWithdrawal() external {
+        // The withdrawal can be finalized 2 rounds for now. We dont make it round + 1 in
+        // case the initiation were to occur right at the end of a round. Doing round + 2 ensures
+        // observers always have at least one full round to become aware of the future balance change.
+        uint64 withdrawalRound = roundTimingInfo.currentRound() + 2;
+        uint256 amount = _balanceOf[msg.sender].balance;
+        _balanceOf[msg.sender].initiateWithdrawal(withdrawalRound);
+        emit WithdrawalInitiated(msg.sender, amount, withdrawalRound);
+    }
+
+    /// @inheritdoc IExpressLaneAuction
+    function finalizeWithdrawal() external {
+        uint256 amountReduced = _balanceOf[msg.sender].finalizeWithdrawal(
+            roundTimingInfo.currentRound()
+        );
+        biddingToken.safeTransfer(msg.sender, amountReduced);
+        emit WithdrawalFinalized(msg.sender, amountReduced);
+    }
+
+    /// @inheritdoc IExpressLaneAuction
+    function flushBeneficiaryBalance() external {
+        uint256 bal = beneficiaryBalance;
+        if (bal == 0) {
+            revert ZeroAmount();
+        }
+        beneficiaryBalance = 0;
+        biddingToken.safeTransfer(beneficiary, bal);
+    }
+
+    /// @dev Update local state to resolve an auction
+    /// @param isMultiBid Where the auction should be resolved from multiple bids
+    /// @param firstPriceBid The winning bid
+    /// @param firstPriceBidder The winning bidder
+    /// @param priceToPay The price that needs to be paid by the winner
+    /// @param biddingInRound The round bidding is taking place in. This is not the round the bidding is taking place for, which is biddingInRound + 1
+    function resolveAuction(
+        bool isMultiBid,
+        Bid calldata firstPriceBid,
+        address firstPriceBidder,
+        uint256 priceToPay,
+        uint64 biddingInRound,
+        RoundTimingInfo memory info
```

Comment:
we can also simply pass in roundStart and roundEnd instead of the struct

Reply:
updated